custom/enikshay/reports/filters.py

```diff
@@ -25,12 +25,29 @@ def options(self):
 
     @property
     def selected(self):
```

Comment:
When a user selects a non-leaf location and he/she hits apply button descendants of selected location will also appear in a select box. That's why I overrode that method to not return descendants. Does it make sense to you?